preview/MsixCore/msixmgr/Package.cpp

```diff
@@ -154,20 +160,64 @@ HRESULT MsixCoreLib::PackageBase::ProcessPSFIfNecessary()
-
-            TraceLoggingWrite(g_MsixTraceLoggingProvider,
-                "PSF redirection",
-                TraceLoggingWideString(m_executionInfo.resolvedExecutableFilePath.c_str(), "Resolved PSF executable"),
-                TraceLoggingWideString(m_executionInfo.workingDirectory.c_str(), "WorkingDirectory"),
-                TraceLoggingWideString(m_executionInfo.commandLineArguments.c_str(), "Arguments"));
+            std::wstring jsonFile = p.path();
+            std::ifstream ifs(jsonFile);
+            IStreamWrapper isw(ifs);
+
+            Document doc;
+            doc.ParseStream(isw);
+
+            if (doc.HasParseError())
+            {
+                //return error
+            }
+
+            const Value& apps = doc["applications"];
+
+            for (Value::ConstValueIterator itr = apps.Begin(); itr != apps.End(); ++itr)
+            {
+                std::string id = (*itr)["id"].GetString();
+                std::wstring jsonApplicationId;
+                jsonApplicationId.assign(id.begin(), id.end());
+
+                //ApplicationId in json should match applicationId from the xml manifest. Right now, we only process first application from the xml manifest anyway
+                if (CaseInsensitiveEquals(jsonApplicationId, m_applicationId))
+                {
+                    std::string executable = (*itr)["executable"].GetString();
+                    std::wstring psfExecutable;
+                    psfExecutable.assign(executable.begin(), executable.end());
+
+                    std::string arguments = (*itr)["arguments"].GetString();
+                    std::wstring psfArguements;
+                    psfArguements.assign(arguments.begin(), arguments.end());
+
+                    std::string workingDirectory = (*itr)["workingDirectory"].GetString();
+                    std::wstring psfWorkingDirectory;
+                    psfWorkingDirectory.assign(workingDirectory.begin(), workingDirectory.end());
+
+                    m_executionInfo.commandLineArguments = psfArguements;
+
+                    // resolve the given paths from json into full paths.
+                    m_executionInfo.resolvedExecutableFilePath = FilePathMappings::GetInstance().GetExecutablePath(psfExecutable, m_packageFullName.c_str());
+                    m_executionInfo.workingDirectory = FilePathMappings::GetInstance().GetExecutablePath(psfWorkingDirectory, m_packageFullName.c_str());
+
+                    const Value& startScript = (*itr)["startScript"];
+                    std::string scriptPath = startScript["scriptPath"].GetString();
+                    std::wstring psfScriptPath;
+                    psfScriptPath.assign(scriptPath.begin(), scriptPath.end());
+                    m_scriptSettings.scriptPath = psfScriptPath;
+
+                    bool showWindow = startScript["showWindow"].GetBool();

```

Comment:
>showWindow [](start = 25, length = 10)

this should default to false if not present in the json

Reply:
Updated to check if member exists in json first. If exists, then set true/false as per value in json file, else set all boolean values to false. By default, if name value pair is not present in json, rapidjson defaults the value of that param to false.